src/postgres_mcp/explain/tools.py

```diff
@@ -0,0 +1,204 @@
+from __future__ import annotations
+
+import logging
+import re
+import json
+from typing import TYPE_CHECKING, Any, Dict, List, Union
+
+from ..dta.artifacts import ExplainPlanArtifact
+
+logger = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+    from ..sql.sql_driver import SqlDriver
+
+
+class ErrorResult:
+    """Simple error result class."""
+
+    def to_text(self) -> str:
+        return self.value
+
+    def __init__(self, message: str):
+        self.value = message
+
+
+class JsonResult:
+    """Simple JSON result class."""
+
+    def to_text(self) -> str:
+        return self.value
+
+    def __init__(self, data: Any):
+        self.value = json.dumps(data, indent=2)
+        self.data = data
+
+
+class ExplainPlanTool:
+    """Tool for generating and analyzing PostgreSQL explain plans."""
+
+    def __init__(self, sql_driver: SqlDriver):
+        self.sql_driver = sql_driver
+
+    async def explain(self, sql_query: str) -> Union[ExplainPlanArtifact, ErrorResult]:
+        """
+        Generate an EXPLAIN plan for a SQL query.
+
+        Args:
+            sql_query: The SQL query to explain
+
+        Returns:
+            ExplainPlanArtifact or ErrorResult
+        """
+        use_generic_plan = self._has_bind_variables(sql_query)
```

Comment:
Note that generic plans only work on Postgres 16+. If possible, error out on older versions. In the future we can add the statistics-based filler.

Reply:
Makes sense. Fixed here: https://github.com/crystaldba/postgres-mcp/pull/23/commits/c7205f17421ef57fc2bd43ac9e0a664ba0182925